### PR TITLE
add amqp_stack_timeout to env vars

### PR DESCRIPTION
**Pull request by shiplet**

### CODE REVIEW

File: great_expectations_cloud/agent/message_service/asyncio_rabbit_mq_client.py [hunk 1]
```diff
@@ -228,8 +229,16 @@ def _on_channel_closed(self, channel: Channel, reason: str) -> None:
         """Callback invoked after the broker closes the channel."""
         self._close_connection()
 
-    def _build_client_parameters(self, url: str) -> pika.URLParameters:
+    def _build_client_parameters(
+        self, url: str, custom_params: dict[str, Any] | None = None
+    ) -> pika.URLParameters:
         """Configure parameters used to connect to the broker."""
+        if custom_params:
+            url_parts = list(parse.urlparse(url))
```
[the remark this answers is not in the document]
Comment (shiplet) [reply great_expectations_cloud/agent/message_service/asyncio_rabbit_mq_client.py:237]: Definitely.

File: great_expectations_cloud/agent/config.py [hunk 2]
```diff
@@ -9,6 +9,7 @@ class GxAgentEnvVars(BaseSettings):  # type: ignore[misc] # BaseSettings is has
     gx_cloud_base_url: AnyUrl = CLOUD_DEFAULT_BASE_URL
     gx_cloud_organization_id: str
     gx_cloud_access_token: str
+    amqp_stack_timeout: int | None = None
```
[the remark this answers is not in the document]
Comment (Kilo59) [reply great_expectations_cloud/agent/config.py:12]: Is this something we want to be configurable by the user? Can we just increase the default timeout slightly and thus avoid all this extra logic (and potential source of user misconfiguration).

File: great_expectations_cloud/agent/message_service/asyncio_rabbit_mq_client.py [hunk 3]
```diff
@@ -228,8 +229,16 @@ def _on_channel_closed(self, channel: Channel, reason: str) -> None:
         """Callback invoked after the broker closes the channel."""
         self._close_connection()
 
-    def _build_client_parameters(self, url: str) -> pika.URLParameters:
+    def _build_client_parameters(
+        self, url: str, custom_params: dict[str, Any] | None = None
```
Comment (Kilo59) [inline great_expectations_cloud/agent/message_service/asyncio_rabbit_mq_client.py:233]:
```suggestion
        self, url: AmqpDsn, custom_params: dict[str, Any] | None = None
```
I don't think we need have this custom query parsing logic.
Pydantic has a `AmqpDsn` type we are alreadying using.
https://docs.pydantic.dev/1.10/usage/types/#urls

We shouldn't be casting the `AmqpDsn` into a string until we are passing it to `pika.URLParameters(str(url))`

File: great_expectations_cloud/agent/message_service/asyncio_rabbit_mq_client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kilo59) [reply great_expectations_cloud/agent/message_service/asyncio_rabbit_mq_client.py:237]: Isn't this just replicating what the `pydantic` `AmqpDsn` already does?

File: great_expectations_cloud/agent/config.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kilo59) [reply great_expectations_cloud/agent/config.py:12]: If we do decide we need to support user provided env configuration we should be using `pydantic` settings management for this.
https://docs.pydantic.dev/1.10/usage/settings/

**Pull request closed without merging**